c_binding/libata.h

```diff
@@ -22,7 +22,50 @@
 #include <stdint.h>
 #include "libstoragemgmt/libstoragemgmt_common.h"
 
-#define _ATA_IDENTIFY_DEVICE_DATA_LEN   512
+#define _ATA_IDENTIFY_DEVICE_DATA_LEN                        512
+
+#define _ATA_REGISTER_INPUT_28_BIT_LENGTH                    7
+#define _ATA_REGISTER_OUTPUT_28_BIT_LENGTH                   7
+#define ATA_PASS_THROUGH_12_LEN                              12
+#define ATA_PASS_THROUGH_12                                  0xa1
+
+/*
+ * ACS-3 7.48.8 SMART RETURN STATUS
+ */
+#define ATA_SMART_RETURN_STATUS_SUBCOMMAND                   0xda
+#define ATA_SMART_COMMAND                                    0xb0
+
+#define ATA_NON_DATA_COMMAND                                 0
+#define ATA_DATA_IN_COMMAND                                  1
+#define ATA_DATA_OUT_COMMAND                                 2
+
+/*
+ * ACS-3 Table 210
+ */
+#define SMART_STATUS_LBA_MID_THRESHOLD_EXCEEDED              0xf4
+#define SMART_STATUS_LBA_HIGH_THRESHOLD_EXCEEDED             0x2c
+#define SMART_STATUS_LBA_MID_DEFAULT                         0x4f
+#define SMART_STATUS_LBA_HIGH_DEFAULT                        0xc2
+
+struct _ata_registers_input_28_bit {
+    uint8_t feature;
+    uint8_t count;
+    uint8_t lba_low;
+    uint8_t lba_mid;
+    uint8_t lba_high;
+    uint8_t device;
+    uint8_t command;
+};
+
+struct _ata_registers_output_28_bit {
+    uint8_t error;
+    uint8_t count;
+    uint8_t lba_low;
+    uint8_t lba_mid;
+    uint8_t lba_high;
+    uint8_t device;
+    uint8_t status;
+};
```

Comment:
You are right. I was reading the ancient ACS-1(I always forgot to delete that PDF).
In both ACS-2 and ACS-3 "Normal and Error Output field descriptions", they are 1 byte.